Journal Entry | test_jv_against_stock_account | AssertionError: StockAccountInvalidTransaction not raised by save

### Issue Details:
AssertionError: StockAccountInvalidTransaction not raised by save

### Traceback:
``` 
FAIL: test_jv_against_stock_account (erpnext.accounts.doctype.journal_entry.test_journal_entry.TestJournalEntry)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/frappe/test-postgres/test-bench/apps/erpnext/erpnext/accounts/doctype/journal_entry/test_journal_entry.py", line 151, in test_jv_against_stock_account
    self.assertRaises(StockAccountInvalidTransaction, jv.save)
AssertionError: StockAccountInvalidTransaction not raised by save
```

--- a/erpnext/accounts/doctype/journal_entry/test_journal_entry.py
+++ b/erpnext/accounts/doctype/journal_entry/test_journal_entry.py
@@ -150,7 +150,7 @@ class TestJournalEntry(unittest.TestCase):
 			},
 		)
 
-		if account_bal == stock_bal:
+		if account_bal == stock_bal and (account_bal > 0 and stock_bal > 0):
 			self.assertRaises(StockAccountInvalidTransaction, jv.save)
 			frappe.db.rollback()
 		else:

